admin/app/main/views/reverify_email.py

```diff
@@ -0,0 +1,64 @@
+import json
+from flask import request, render_template, session, abort, current_app, flash, redirect, url_for
+from flask_login import current_user
+from notifications_utils.url_safe_token import check_token
+from itsdangerous import SignatureExpired
+from app import user_api_client
+from app.main import main
+from app.utils import redirect_to_sign_in
+
+
+@main.route('/reverify-email-token/<token>')
+def reverify_email_token(token):
+    try:
+        token_data = check_token(
+            token,
+            current_app.config['SECRET_KEY'],
+            current_app.config['DANGEROUS_SALT'],
+            current_app.config['EMAIL_EXPIRY_SECONDS']
+        )
+    except SignatureExpired:
+        flash('The link in the email we sent you has expired. We\'ve sent you a new one.')
+        return redirect(url_for('main.resend_email_reverification'))
+
+    token_data = json.loads(token_data)
+    user = user_api_client.get_user(token_data['user_id'])
+
+    if not user:
+        abort(404)
+
+    session['user_details'] = {
+        'email': user.email_address,
+        'id': user.id,
+        'email_reverified': True,
+    }
+
+    user_api_client.send_verify_code(user.id, 'sms', user.mobile_number)
```

Comment:
same as comment on other ticket about authentication methods

Reply:
I've updated this to exclude email auth users from the email reverification check - my thinking is that each time they sign in with email 2fa they are already reverifying. Does that make sense?

I have also made sure that the email_last_verified_at field is also being updated during email 2fa.